lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java

```diff
@@ -792,17 +793,32 @@ public void deleteFiles(Collection<Path> paths)
     }
 
     private void deletePaths(List<Path> paths)
+            throws IOException
     {
+        String bucketName = getBucketName(uri);
         List<KeyVersion> keys = paths.stream()
                 .map(TrinoS3FileSystem::keyFromPath)
                 .map(KeyVersion::new)
                 .collect(toImmutableList());
-        DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(getBucketName(uri))
+        DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(bucketName)
                 .withRequesterPays(requesterPaysEnabled)
                 .withKeys(keys)
                 .withQuiet(true);
 
-        s3.deleteObjects(deleteObjectsRequest);
+        try {
+            s3.deleteObjects(deleteObjectsRequest);
+        }
+        catch (MultiObjectDeleteException e) {
```

Comment:
The caller method catches this exception: https://github.com/trinodb/trino/commit/7a863b30c1448024feb92bbd6119943d977f80b5 (by the last week change)